File: akka-java/src/main/java/org/hahn/spider/actor/BuildIndex.java
package org.hahn.spider.actor;

import akka.actor.UntypedActor;
import org.hahn.spider.message.Index;

/**
 * Created by jianghan on 16/5/2.
 */
public class BuildIndex extends UntypedActor {

    @Override
    public void onReceive(Object msg) throws Exception {
        if (msg instanceof Index) {
            Index index = (Index) msg;
            System.out.println(index);
        } else {
            unhandled(msg);
        }
    }

}

File: akka-scala/src/main/scala/hahn/learning/akka/spider/actor/BuildIndex.scala
package hahn.learning.akka.spider.actor

import akka.actor.Actor
import hahn.learning.akka.spider.Message

/**
  * Created by jianghan on 16-5-26.
  */
class BuildIndex extends Actor {

  def receive = {
    case Message.Index(t, url, title, content) => println(s"${t}\t${url}\t${title}\t${content}")
    case _ => println("ERROR BuildIndex")
  }

}
